### PR TITLE
Feat/add zeroex nft table

### PR DESCRIPTION
**Pull request by bakabhai993**
Brief comments on the purpose of your changes:

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [x] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [x] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by bakabhai993** [comment]
@Hosuke, is there anything else I need to do? I am trying to make changes to this repository for the first time.

I just wanted to make sure I understood the commit. "Merge branch 'main' into pr/2175"

**Comment by Hosuke** [comment]
> I just wanted to make sure I understood the commit. "Merge branch 'main' into pr/2175"

This commit is to sync with the latest main branch, so it updates all latest commits into this PR/branch.
However, it seems this PR currently currently is unable to pass the dbt-test, which is an issue.
We need to try to make this PR pass the dbt-test so it can be safely merged into main branch.

**Comment by Hosuke** [comment]
I am currently trying to fix [this error](https://github.com/duneanalytics/spellbook/actions/runs/3674573350/jobs/6212951515). However I do not have any clue/idea at the moment.
One possible cause may be the changes in file structures conflict with the dbt_project.yml.

File: dbt_project.yml [hunk 1]
```diff
@@ -981,26 +985,7 @@ seeds:
...
-            taker_token_amount: double
-            taker_token_amount_raw: double
-            maker_token_amount: double
-            maker_token_amount_raw: double
-            type: string
-            affiliate_address: string
-            swap_flag: boolean
-            matcha_limit_order_flag: boolean
-            volume_usd: double
+        # TODO define the new zeroex seed files here
```
[the remark this answers is not in the document]
Comment (bakabhai993) [reply dbt_project.yml:988]: Thank you for pointing this out.

[the remark this answers is not in the document]
Comment (bakabhai993) [reply dbt_project.yml]: @Hosuke i have made the changes, but it is still failing for some reason.

**Comment by bakabhai993** [comment]
btw, i have make changes in the folder structure and the schema for an already existing model , could that be the source of the issue?
@Hosuke

File: dbt_project.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hosuke) [reply dbt_project.yml]: I will check again maybe tomorrow.

**Comment by Hosuke** [comment]
> btw, i have make changes in the folder structure and the schema for an already existing model , could that be the source of the issue? @Hosuke

The folder structure you changed need to be reflected in the `dbt_project.yml`, so the dbt will know the linking/mapping in the building stage. You can see `dbt_project.yml` is a summary of directory structures of model/seed/test in the project.

**Comment by bakabhai993** [comment]
@Hosuke I was making a typo in the [dbt_project.yml] file, i have corrected the issue now.

**Comment by bakabhai993** [comment]
what will be the next steps in this case?

**Comment by Hosuke** [comment]
> what will be the next steps in this case?

I will try to make sure `check_seed` tests correctly applied and check for incremental settings.
And after that mark this PR ready for final review.

**Comment by Hosuke** [comment]
This PR moves original `zeroex_api_ethereum_fills` to `erc20` folder, and add a new model `zeroex_nft_ethereum_fills`.

File: seeds/zeroex/ethereum/fills/zeroex_api_ethereum_fills_sample.csv [hunk 2]
```diff
@@ -0,0 +1,6 @@
+tx_hash,block_number,evt_index,contract_address,block_time,maker,taker,taker_token,taker_symbol,maker_token,maker_symbol,token_pair,taker_token_amount,taker_token_amount_raw,maker_token_amount,maker_token_amount_raw,type,affiliate_address,swap_flag,matcha_limit_order_flag,volume_usd,tx_from,tx_to
+0x4ad0da0192cf45284124b5a480fafed17f89e5474412dedf03cac86bfab980db,1.6068182e+07,101,0x1d42064fc4beb5f8aaf85f4617ae8b3b5b8bd801,2022-11-28T11:27:23Z,0x1d42064fc4beb5f8aaf85f4617ae8b3b5b8bd801,0xd3a8813e58d29e4b9c126bd80ebae12a415b3e55,0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2,WETH,0x1f9840a85d5af5bf1d1762f925bdaddc4201f984,UNI,UNI-WETH,7.52817337639159,7.52817337639159E18,1667.3526432935869,1.667352643293587E21,Uniswap V3 Direct,0x720c9244473dfc596547c1f7b6261c7112a3dad4,true,false,8831.60131478003,0xd3a8813e58d29e4b9c126bd80ebae12a415b3e55,0xdef1c0ded9bec7f1a1670819833240f027b25eff
```
Comment (Hosuke) [inline seeds/zeroex/ethereum/fills/zeroex_api_ethereum_fills_sample.csv:2]: The `block_number` should be an int-typed number in my opinion.

**Review by Hosuke — APPROVED** [review]
To whom it may concern, I have reviewed this PR and here is the summary:

- Metadata & naming conventions look good ✅
- Config blocks match uniqueness tests in schema files ✅
- Tests and seeds added for the new models ✅
- Incremental filters added when joining large tables ✅
- Contract addresses are lowercased ✅ 

[Sample query results here](https://dune.com/queries/1757282)

**Comment by bakabhai993** [comment]
Dear @jeff-dude ,

I hope this message finds you well. Could you please let me know if there are any next steps that need to be taken on our end?

Sincerely,
@bakabhai993

**Comment by dot2dotseurat** [comment]
@bakabhai993 can you resolve the conflicts on the dbt_project.yml file? I'm happy to merge once that's resolved.

**Pull request merged**